packages/webapp/src/components/Charts/LineChart.tsx

```diff
@@ -0,0 +1,221 @@
+import type { NameType, ValueType } from 'recharts/types/component/DefaultTooltipContent';
+import type { DataKey } from 'recharts/types/util/types';
+import { Payload } from 'recharts/types/component/DefaultLegendContent';
+import { getDateTime, getLocalShortDate } from './utils';
+import styles from './styles.module.scss';
+
+interface LineConfig {
+  id: string;
+  color: string;
+}
+
+export type ChartTruncPeriod = 'day' | 'hour';
+
+interface CommonProps {
+  title?: string;
+  lineConfig: LineConfig[];
+  formatTooltipValue?: (label: any, value?: ValueType) => string | number;
+  isCompactView?: boolean;
+}
+
+type TimeScaleProps = CommonProps & {
+  xAxisDataKey?: 'dateTime';
+  data: { dateTime: number; [key: string]: string | number | undefined | null }[];
+  language: string;
+  truncPeriod: ChartTruncPeriod;
+  ticks?: number[];
+};
+
+type GeneralScaleProps<T extends string> = CommonProps & {
+  xAxisDataKey: T;
+  data: ({ [K in T]: string | number } & { [key: string]: string | number | undefined | null })[];
+  ticks?: number[] | string[];
+};
+
+export type LineChartProps<T extends string = never> = TimeScaleProps | GeneralScaleProps<T>;
+
+const DATE_TIME = 'dateTime';
+
+function isTimeScaleProps(props: LineChartProps): props is TimeScaleProps {
+  return !props.xAxisDataKey || props.xAxisDataKey === DATE_TIME;
+}
+
+const axisLabelStyles = {
+  stroke: '#2B303A',
+  strokeWidth: 1,
+  fontFamily: 'Open Sans',
+  fontSize: '12px',
+  letterSpacing: '0.15px',
+};
+
+/**
+ * Renders a responsive line chart using Recharts.
+ *
+ * Uses the `activeLine` state to override the default Recharts behaviour of displaying
+ * tooltips and dots for all x-axis points, ensuring that tooltip values and dots are
+ * shown only for the hovered line.
+ */
+function LineChart(props: LineChartProps) {
+  const {
+    title,
+    data,
+    lineConfig,
+    ticks,
+    xAxisDataKey = DATE_TIME,
+    formatTooltipValue,
+    isCompactView,
+  } = props;
+
+  const { t } = useTranslation('common');
+
+  const [activeLine, setActiveLine] = useState('');
+  const [hiddenLines, setHiddenLines] = useState<string[]>([]);
+
+  const showLegend = lineConfig.length > 1;
+
+  const CustomTooltip = ({ active, payload, label }: TooltipProps<ValueType, NameType>) => {
+    if (active && payload?.length && activeLine) {
+      const data = payload.find(({ dataKey }) => dataKey === activeLine);
+
+      if (data) {
+        const { color } = lineConfig.find(({ id }) => id === activeLine)!;
+        const xAxisData = isTimeScaleProps(props)
+          ? getDateTime(label, props.language, props.truncPeriod, t)
+          : label;
+        const value = formatTooltipValue?.(label, data.value) || data.value;
+
+        return (
+          <dl className={styles.tooltip} style={{ '--tooltipColor': color } as React.CSSProperties}>
+            <dt>{xAxisData}</dt>
+            <dd>{value}</dd>
+          </dl>
+        );
+      }
+    }
+
+    return null;
+  };
```

Comment:
Could we move this component outside of `LineChart` and pass any props needed from LineChart to it? It's best not to declare a component inside another one, because it will be re-created on every render. I also think it's harder to figure out which data the component needs to function when it's declared inside another one and uses some of its props/state